src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.cs

```diff
@@ -1904,6 +1904,25 @@ public async Task<bool> ContainsSymbolsWithNameAsync(ProjectId id, Func<string,
             return compilation.ContainsSymbolsWithName(predicate, filter, cancellationToken);
         }
 
+        internal async Task<INamespaceDeclaration> GetDeclarationRootAsync(ProjectId id, CancellationToken cancellationToken)
+        {
+            var result = GetCompilationTracker(id).GetDeclarationRootFromDeclarationOnlyCompilation();
+            if (result != null)
+            {
+                return result;
+            }
+
+            // it looks like declaration compilation doesn't exist yet. we have to build full compilation
+            var compilation = await GetCompilationAsync(id, cancellationToken).ConfigureAwait(false);
+            if (compilation == null)
+            {
+                // some projects don't support compilations (e.g., TypeScript) so there's nothing to check
+                return null;
+            }
```

Comment:
this seems odd and unfortunate.  i'd much rather have us just produce the decl-only compilation and jsut use that insead of building the full compilation.